src/containers/HomePage.js

```diff
@@ -4,31 +4,91 @@ import Menu from 'components/common/Menu';
 import LogoutButton from 'components/user/LogoutButton';
 import SimpleMap from 'components/common/maps/Map';
 import Welcome from 'components/user/Welcome';
+import CreateTargetForm from 'components/user/CreateTargetForm';
 import { array, func } from 'prop-types';
 
-import { loadTargets } from '../actions/targetActions';
+import { loadTargets, addTarget } from '../actions/targetActions';
 import { loadTopics } from '../actions/topicActions';
+import { FormattedMessage } from 'react-intl';
 
 class HomePage extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      targetPosition: {},
+      isCreatingNewTarget: false,
+    };
+    this.onClickMap = this.onClickMap.bind(this);
+    this.handleCreateTarget = this.handleCreateTarget.bind(this);
+  }
+
   componentDidMount() {
     this.props.loadTargets();
     this.props.loadTopics();
   }
 
+  onClickMap = ({ x, y, lat, lng, event }) => {
+    console.log(x, y, lat, lng, event);
+    const targetPosition = {
+      lat,
+      lng
+    };
+    this.setState({ targetPosition });
+    this.setState({ isCreatingNewTarget: true });
+  }
+
+  handleCreateTarget(data) {
+    const { title, radius, topicId } = data.toJS();
+    const { lat, lng } = this.state.targetPosition;
+    const targetCompleted = {
+      target: {
+        lat,
+        lng,
+        title,
+        radius,
+        topic_id: topicId[0]
+      }
+    };
+    this.props.addTarget(targetCompleted);
+  }
+
+  MenuLeft(topicList) {
+    let menu;
+    if (this.state.isCreatingNewTarget) {
```

Comment:
Also, you can remove the `let` declaration and return rightaway with a ternary condition like `condition ? something : somethingelse`

````
return this.state.isCreatingNewTarget ?
  whatever :
  whatever2
```